File: baekjoon/11404.py
import sys
input = sys.stdin.readline
n = int(input())    # n 도시 개수
m = int(input())    # m 버스 개수
lst = [[] for _ in range(n)]

INF = 21e8
dist = [[INF] * n for _ in range(n)]

for _ in range(m):
    st, ed, cost = map(int, input().split())
    dist[st-1][ed-1] = min(dist[st-1][ed-1], cost)
for i in range(n):
    for j in range(n):
        if i == j: dist[i][j] = 0

for k in range(n):
    for i in range(n):
        for j in range(n):
            dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])

for i in range(n):
    for j in range(n):
        if dist[i][j] == INF:
            print(0, end=" ")
        else:
            print(dist[i][j], end=" ")
    print()

File: baekjoon/11404.js
// https://www.acmicpc.net/problem/11404
// 플로이드
// 플로이드 워셜

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const node = Number(input.shift());
const n = Number(input.shift());
const route = input.map((v) => v.split(" ").map(Number));

function solution() {
  const INF = 21e8;
  let dist = Array.from(Array(node), () => Array(node).fill(INF));
  route.forEach(
    ([start, end, cost]) =>
      (dist[start - 1][end - 1] = Math.min(dist[start - 1][end - 1], cost))
  );

  for (let i = 0; i < node; i++) {
    dist[i][i] = 0;
  }

  for (let d = 0; d < node; d++) {
    for (let i = 0; i < node; i++) {
      for (let j = 0; j < node; j++) {
        dist[i][j] = Math.min(dist[i][j], dist[i][d] + dist[d][j]);
      }
    }
  }
  dist.forEach((col) => {
    for (let v in col) {
      if (col[v] == INF) col[v] = 0;
    }
    console.log(...col);
  });
}
solution();
